### DISCUSSION TITLE
Efficiency reading text file

### DISCUSSION THREAD

**Discussion by lfcc1** [explosion/spaCy#8792] [Help: Coding & Implementations] [answered]
Greetings,
Im tryng to extrac entities from an unstructured text file. 
I have used two differente implementations:

```
file = open("./example, encoding='utf-8')
for line in file:
    doc = nlp(line)
    displacy.render(doc, style="ent") 
```

----------------------------------

```
file = open("./example",encoding='utf-8').read()
doc = nlp(file)
displacy.render(doc, style="ent") 
```
The first approach is much slower but it processes the file line by line.
The second approach reads the whole file but, isn't it a problem to read the whole file at once?
Is there a better way to do this?

Thank you.

**Comment by polm** [comment, marked as answer]
Please see the [speed FAQ](https://github.com/explosion/spaCy/discussions/8402). 

Generally you would want to do something like this:

```
file = open("blah")
lines = (line for line in file)
for doc in nlp.pipe(lines):
    ... do stuff ...
```

You can also make it faster by turning off components you don't use. Since you're using the entities you could disable the parser and tagger, for example. 

> The second approach reads the whole file but, isn't it a problem to read the whole file at once?

Why would it be a problem? There are reasons that can be bad (like if the file doesn't fit in memory), but without knowing more about what you're doing I'm not sure why that would be the case. (Rendering a whole file in one displaCy call might be hard to read...?)
